src/Evolution/Systems/ScalarAdvection/FiniteDifference/ReconstructWork.tpp

```diff
@@ -0,0 +1,161 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <boost/functional/hash.hpp>
+#include <cstddef>
+#include <utility>
+
+#include "DataStructures/FixedHashMap.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/DirectionMap.hpp"
+#include "Domain/Structure/Element.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Structure/MaxNumberOfNeighbors.hpp"
+#include "Evolution/DgSubcell/NeighborData.hpp"
+#include "Evolution/Systems/ScalarAdvection/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace ScalarAdvection::fd {
+template <size_t Dim, typename TagsList, typename Reconstructor>
+void reconstruct_work(
+    const gsl::not_null<std::array<Variables<TagsList>, Dim>*>
+        vars_on_lower_face,
+    const gsl::not_null<std::array<Variables<TagsList>, Dim>*>
+        vars_on_upper_face,
+    const Reconstructor& reconstruct,
+    const Variables<tmpl::list<Tags::U>> volume_vars,
+    const Element<Dim>& element,
+    const FixedHashMap<maximum_number_of_neighbors(Dim) + 1,
+                       std::pair<Direction<Dim>, ElementId<Dim>>,
+                       evolution::dg::subcell::NeighborData,
+                       boost::hash<std::pair<Direction<Dim>, ElementId<Dim>>>>
+        neighbor_data,
+    const Mesh<Dim>& subcell_mesh, const size_t ghost_zone_size) {
+  // check if subcell mesh is isotropic
+  ASSERT(Mesh<Dim>(subcell_mesh.extents(0), subcell_mesh.basis(0),
+                   subcell_mesh.quadrature(0)) == subcell_mesh,
+         "The subcell mesh should be isotropic but got " << subcell_mesh);
+
+  // number of points at which reconstruction is performed for each dimension,
+  // which equals the number of interfaces.
+  const size_t reconstructed_num_pts =
+      (subcell_mesh.extents(0) + 1) *
+      subcell_mesh.extents().slice_away(0).product();
+
+  // make span for return values
+  std::array<gsl::span<double>, Dim> upper_face_vars{};
+  std::array<gsl::span<double>, Dim> lower_face_vars{};
+  for (size_t i = 0; i < Dim; i++) {
+    gsl::at(upper_face_vars, i) =
+        gsl::make_span(get<Tags::U>(gsl::at(*vars_on_upper_face, i))[0].data(),
+                       reconstructed_num_pts);
+    gsl::at(lower_face_vars, i) =
+        gsl::make_span(get<Tags::U>(gsl::at(*vars_on_lower_face, i))[0].data(),
+                       reconstructed_num_pts);
+  }
+
+  // number of ghost FD cells for each Direction
+  const size_t neighbor_num_pts =
```

Comment:
would naming this `neighbor_num_ghost_fd_points` be clear enough to remove the comment? I prefer clearly named variables over comments when possible (not always possible, though!)